src/python/pants/option/custom_types.py

```diff
@@ -15,6 +15,13 @@
 from pants.util.strutil import ensure_text
 
 
+# A default value for a bool typed option that indicates it can act as a tri-state that defaults to
+# `None`; ie: the option is un-set.
+#
+# :API: public
+UnsetBool = object()
```

Comment:
Eh, yeah - I guess. This is dancing around docstrings though. That wart fixed, the new wart is you can do UnsetBool() which we don't actually want.